src/pyotp/otp.py

```diff
@@ -18,6 +18,8 @@ def __init__(
         issuer: Optional[str] = None,
     ) -> None:
         self.digits = digits
+        if digits > 10:
+            raise ValueError("digits must be no greater than 10")
```

Comment:
This introduces a breaking limitation. It is needed because of the hardcoded `10_000_000_000` below. See there for a suggestion of how to deal with this.

Reply:
The limitation already existed. It comes from the fact that the HMAC is only 32 bits, so if you try to extract more than 10 digits you'll only get extra zeroes.

The proper limit is actually 9 digits, but the Steam subclass needs 10, so I allowed 10.